Http disruptor (#16)
grafana/k6-chaos: README.md

`kill`: kill pod

`slowdownNetwork`: delays network traffic for the pod

    Parameters:
      options: controls the delay attack
        - delay: average delay in network packages (in milliseconds)
        - variation: variation in the delay (in milliseconds)
        - duration: duration of the disruption 

`disruptHttp`: disrupts http requests to the pod.

      Parameters:
        options: controls the attack

>           - delay: average delay in network packages (in milliseconds)

Reviewer: ```suggestion
          - delay: average delay in requests (in milliseconds)
```

Author: Good catch. Thanks,

          - variation: variation in the delay (in milliseconds)
          - error_rate: rate of requests that will return an error
          - error_code: error code to return
          - duration: duration of the disruption
          - target: port on which the requests will be intercepted (defaults to 80)
          - port: port the transparent proxy will use to listen for requests (defaults to 8080)
          - interface: interface on thich the traffic will be intercepted (defaults to eth0)

## Examples

> The [./exaples](./examples) forlder contains examples of using `k6-chaos`. 

Reviewer: ```suggestion
The [./examples](./examples) folder contains examples of using `k6-chaos`. 
```

```

## Introduce disruptions in http connections to pod

The [disrupt-http.js example](examples/disrupt-http.js) shows how `PodDisruptor` can be used for testing the effect of disruptions in the http requests served by an instance of a deloyment. The example uses a deployment of the `httpbin`. The initial setup makes each request to take `100ms`. When executed without any disruption the statistics looks like this.

```
http_req_duration..............: avg=101.66ms min=100.59ms med=101.69ms max=107.24ms p(90)=102.22ms p(95)=102.36ms
```

> When the disruption of an additional `100ms` delay and an error rate of 10% of requests is introduced for a period of `30s`, we can see how the stastics are affected (in particular, `p(90)` and `p(95)`):

Reviewer: ```suggestion
When the disruption of an additional `100ms` delay and an error rate of 10% of requests is introduced for a period of `30s`, we can see how the statistics are affected (in particular, `p(90)` and `p(95)`):
```
